cmake/catkin_package.cmake

```diff
@@ -400,12 +401,14 @@ function(_catkin_package)
   set(DEVELSPACE FALSE)
   set(INSTALLSPACE TRUE)
 
-  set(PROJECT_SPACE_DIR ${CMAKE_INSTALL_PREFIX})
+  set(PROJECT_SPACE_LIBDIR ${CMAKE_INSTALL_PREFIX}/${CATKIN_GLOBAL_LIB_DESTINATION})
+  set(PROJECT_SPACE_DATADIR ${CMAKE_INSTALL_PREFIX}/${CATKIN_GLOBAL_SHARE_DESTINATION})
```

Comment:
This suffixing means that if you set the the GNUInstallDirs vars to absolute paths, you have to set `CMAKE_INSTALL_PREFIX` to `/`. If we don't like that, then we could also put a check in here so that if the `CATKIN_GLOBAL_X_DESTINATION` vars are absolute, they'll be used as-is.

Reply:
Upon reflection, this is not at all workable. A generic build framework needs to be able to set both the GNUInstallDirs vars and also the generic prefix one, so that both types of packages can be expected to do the right thing. Will add the absolute directory check.